pandas/core/window/rolling.py

```diff
@@ -1185,14 +1185,16 @@ def count(self):
 
         window = self._get_window()
         window = min(window, len(obj)) if not self.center else window
+        min_periods = self.min_periods if self.min_periods is not None else 0
+        min_periods = min(min_periods, len(obj)) if not self.center else min_periods
 
         results = []
         for b in blocks:
             result = b.notna().astype(int)
             result = self._constructor(
                 result,
                 window=window,
-                min_periods=0,
+                min_periods=min_periods,
```

Comment:
What is the downside of just passing `self.min_periods` here? Would be more consistent with other places in the module

Reply:
hi Will, this is for backward compatibility purpose. If we set `min_periods=self.min_periods`, many (63 to be exact) existing test cases fail, which all assume `min_periods` is 0. 

Ping me if you think we should break this backward-compatibility in favor of consistency with other APIs.

Reply:
Are the tests wrong? Seems like we should be consistent

Reply:
I just realize the real reason behind this design (making ``min_periods=0``): because ``min_periods`` is interpreted as "minimum number of valid (i.e. not null) entries" rather than "minimum number of entries in the current window".

To illustrate what I mean:
```python
>>> s = pd.Series([1, 2, 3, np.nan, 4, 5, 6])
>>> s
0    1.0
1    2.0
2    3.0
3    NaN
4    4.0
5    5.0
6    6.0
dtype: float64

>>> s.rolling(3, min_periods=3).mean()  # note we have NaNs for iloc[3:6]
0    NaN
1    NaN
2    2.0
3    NaN
4    NaN
5    NaN
6    5.0
dtype: float64
```

Now, the exact point of `count` method is to count how many values are not NaNs, if we make this exactly the same as other APIs, i.e. by default `min_periods=None` which will then be assigned to be `min_periods = window` inside the functions, then we would give NA as long as there is at least one NA in the window, or full window length otherwise, like this:
```python
# This would be the result if we make rolling count behave the same as other APIs. 
# I feel this should not be the default behavior
>>> s.rolling(3).count()  
0    NaN
1    NaN
2    3.0
3    NaN
4    NaN
5    NaN
6    3.0

# This is what I think it should be
>>> s.rolling(3).count()
0    1.0
1    2.0
2    3.0
3    2.0
4    2.0
5    2.0
6    3.0

# However I do want to allow the user to specify min_periods 
# and make the function behave like the others
>>> s.rolling(3, min_periods=3).count()  
0    NaN
1    NaN
2    3.0
3    NaN
4    NaN
5    NaN
6    3.0
```
The last case is what this PR is trying to achieve.